Mathlib/Algebra/Order/BigOperators/Group/LocallyFinite.lean

```diff
@@ -3,78 +3,126 @@ Copyright (c) 2024 Yaël Dillies. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yaël Dillies
 -/
-import Mathlib.Order.Interval.Finset.Basic
-import Mathlib.Algebra.BigOperators.Group.Finset.Basic
+import Mathlib.Algebra.BigOperators.Group.Finset.Sigma
+import Mathlib.Algebra.Order.Interval.Finset.SuccPred
+import Mathlib.Data.Nat.SuccPred
+import Mathlib.Order.Interval.Finset.Nat
 
 /-!
 # Big operators indexed by intervals
 
 This file proves lemmas about `∏ x ∈ Ixx a b, f x` and `∑ x ∈ Ixx a b, f x`.
 -/
 
-variable {α β : Type*} [PartialOrder α] [CommMonoid β] {f : α → β} {a b : α}
+open Order
+
+variable {α M : Type*} [CommMonoid M] {f : α → M} {a b : α}
 
 namespace Finset
+section PartialOrder
+variable [PartialOrder α]
+
 section LocallyFiniteOrder
 variable [LocallyFiniteOrder α]
 
 @[to_additive]
-lemma left_mul_prod_Ioc (h : a ≤ b) : f a * ∏ x ∈ Ioc a b, f x = ∏ x ∈ Icc a b, f x := by
-  rw [Icc_eq_cons_Ioc h, prod_cons]
+lemma mul_prod_Ico_eq_prod_Icc (h : a ≤ b) : f b * ∏ x ∈ Ico a b, f x = ∏ x ∈ Icc a b, f x := by
+  rw [Icc_eq_cons_Ico h, prod_cons]
 
-@[to_additive]
-lemma prod_Ioc_mul_left (h : a ≤ b) : (∏ x ∈ Ioc a b, f x) * f a = ∏ x ∈ Icc a b, f x := by
-  rw [mul_comm, left_mul_prod_Ioc h]
+@[deprecated (since := "2025-05-03")] alias right_mul_prod_Ico := mul_prod_Ico_eq_prod_Icc
 
 @[to_additive]
-lemma right_mul_prod_Ico (h : a ≤ b) : f b * ∏ x ∈ Ico a b, f x = ∏ x ∈ Icc a b, f x := by
-  rw [Icc_eq_cons_Ico h, prod_cons]
+lemma prod_Ico_mul_eq_prod_Icc (h : a ≤ b) : (∏ x ∈ Ico a b, f x) * f b = ∏ x ∈ Icc a b, f x := by
+  rw [mul_comm, mul_prod_Ico_eq_prod_Icc h]
 
-@[to_additive]
-lemma prod_Ico_mul_right (h : a ≤ b) : (∏ x ∈ Ico a b, f x) * f b = ∏ x ∈ Icc a b, f x := by
-  rw [mul_comm, right_mul_prod_Ico h]
+@[deprecated (since := "2025-05-03")] alias prod_Ico_mul_right := prod_Ico_mul_eq_prod_Icc
 
 @[to_additive]
-lemma left_mul_prod_Ioo (h : a < b) : f a * ∏ x ∈ Ioo a b, f x = ∏ x ∈ Ico a b, f x := by
-  rw [Ico_eq_cons_Ioo h, prod_cons]
+lemma mul_prod_Ioc_eq_prod_Icc (h : a ≤ b) : f a * ∏ x ∈ Ioc a b, f x = ∏ x ∈ Icc a b, f x := by
+  rw [Icc_eq_cons_Ioc h, prod_cons]
 
-@[to_additive]
-lemma prod_Ioo_mul_left (h : a < b) : (∏ x ∈ Ioo a b, f x) * f a = ∏ x ∈ Ico a b, f x := by
-  rw [mul_comm, left_mul_prod_Ioo h]
+@[deprecated (since := "2025-05-03")] alias left_mul_prod_Ioo := mul_prod_Ioc_eq_prod_Icc
```

Comment:
This looks like a very confusing deprecation: the old one was about multiplying into an Ioo to make an Ico, but the proposed replacement multiplies into an Ioc to make an Icc.

Reply:
Thanks for the fresh pair of eyes, this was a weird PR to keep afloat, and not everything stayed as intended